doc/working-stealth-jobs.md

```diff
@@ -1,3 +1,84 @@
 # Working Stealth Jobs
 
-TBD
+**[YearnStealthVault](https://etherscan.io/address/0xYearnStealthVault#code)**
+
+**[YearnStealthRelayer](https://etherscan.io/address/0xYearnStealthRelayer#code)**
+
+
+
+### How to bond:
+
+- to add bonds you need to call (bond) with as much ETH (msg.value) as you'd like to bond.
+    - check amount being bonded is higher than the penalty of the job you want to perform work on
+
+- call `enableStealthContract(address _contract)` with `_contract` being the address of the contract you want to perform work on
+    - in this case `0xYearnStealthRelayer` (YearnStealthRelayer)
+    - you can also use `enableStealthContracts(address[] calldata _contracts)`
+    - this is required since contracts have the ability to take your bond away.
+        - so, please be careful and only enable contracts after reviewing them.
+
+- to remove a contract from your list just call `disableStealthContract(address _contract)`
+    - you can also use `disableStealthContracts(address[] calldata _contracts)`
+
+- to unbond, call (startUnbond), wait 4 days, and call (unbond | unbondAll).
+    - in this period you'll not be able to perform any work
+    - to cancel unbond and continue working call (cancelUnbond)
+
+### How to work:
+
+- here is an example on how to call a job though `StealthRelayer`:
+```ts
+    // First, grab the data for the underlying tx
+    const rawTx = await stealthERC20.connect(alice).populateTransaction.stealthMint(alice.address, mintAmount);
+    const callData = rawTx.data;
+
+    // then generate a random string and compact it into 32 bytes
+    const stealthHash = ethers.utils.solidityKeccak256(['string'], ['random-secret-hash']);
+    // then get the block number (useful for bandit attack protection)
+    let blockNumber = await ethers.provider.getBlockNumber();
+
+    // and finally execute your tx. (this step can be changed to use flashbots, see guide below)
+    await stealthRelayer.connect(alice).execute(
+      stealthERC20.address, // address _job,
+      callData, // bytes memory _callData,
+      stealthHash, // bytes32 _stealthHash,
+      blockNumber + 1 // uint256 _blockNumber
+    );
+```
+
+### Risks:
+
+- at any point in time governance can take your bond away.
+- if the private mempool service you were using leaks your tx, you can get your bond slashed.
+    - you still might be able to recover up to 90% of the penalty if there was no harm done.
+
+
+### Requirements:
+
+- interacting with the StealthRelayer **MUST ALWAYS** be done using an EOA.
+    - failure to comply with this requirement will result in instant and unreversable slashing of all your bond.
```

Comment:
This is not automatically, right ?

Reply:
right now we can force EOA check using the `block.gasLimit ` trick. but it might become obsolete in the future, better to leave it as is.

BTW, i need to update the docs to reflect the need to send as much gas as `block.gasLimit`

Reply:
@lbertenasco don't forget about that on the docs. :yum:

Reply:
added below :)